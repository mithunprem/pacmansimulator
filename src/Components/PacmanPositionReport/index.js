import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const PacmanPositionReport = ({ pacmanPosition }) => {

  const { x, y, direction } = pacmanPosition;

  return (
    <Fragment>
      <div className="report alert alert-info">
        <span>Pacman's current position is: {x}, {y}, {direction.toUpperCase()}</span>
      </div>
    </Fragment>
  );
}

PacmanPositionReport.propTypes = {
  pacmanPosition: PropTypes.object
}

export default PacmanPositionReport;
