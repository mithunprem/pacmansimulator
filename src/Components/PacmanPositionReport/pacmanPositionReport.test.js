import React from 'react';
import { create } from 'react-test-renderer';
import PacmanPositionReport from '../../Components/PacmanPositionReport';

describe("PacmanPositionReport", () => {
  const pacmanPosition = { x:1, y:1, direction:'NORTH' };
  test("should render without throwing an error", () => {
    const report = create(<PacmanPositionReport pacmanPosition={pacmanPosition} />).toJSON()
    expect(report).toMatchSnapshot()
  });
});
