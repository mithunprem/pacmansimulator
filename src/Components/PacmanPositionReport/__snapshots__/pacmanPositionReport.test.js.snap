// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PacmanPositionReport should render without throwing an error 1`] = `
<div
  className="report alert alert-info"
>
  <span>
    Pacman's current position is: 
    1
    , 
    1
    , 
    NORTH
  </span>
</div>
`;
